### ISSUE TITLE
K.sqrt cause loss is nan in CPU, GPU is correct

### ISSUE THREAD

**Issue by Flyfoxs** [keras-team/keras#13203]
**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  
- TensorFlow backend (yes / no):  Yes
- TensorFlow version: 1.10.1
- Keras version:  2.1.6
- Python version:  3.6
- CUDA/cuDNN version:  
- GPU model and memory:  CPU

**Describe the current behavior**  
loss is nan
**Describe the expected behavior**  
loss is a valueable number

**Code to reproduce the issue**  
Here is the code, you can see the loss is nan in **CPU**,  but if you remove the line with sqrt it  will be correct.
from keras.layers import Dense
from keras.models import load_model, Sequential
from keras.utils import CustomObjectScope
import tensorflow as tf
import keras as keras
from keras import backend
from keras.models import Model
from keras.optimizers import Adam
from keras import losses 

def get_test_mode():
    model = Sequential()
    from keras.layers import Flatten
    input = keras.Input(shape=(10,10), dtype='float', name='raw_image_left')
    test = keras.layers.Dense(units=64, activation='relu', name='middle', )(input)
    test = keras.layers.Flatten()(test)
    #Comments this line will be correct, which is only  incorrect in CPU
    test = keras.layers.Lambda(lambda x: keras.backend.sqrt(x), name='error_point')(test)
    output = keras.layers.Dense(units=1, activation='sigmoid', name='output', )(test)
    model = Model(input, output)
    return model

model = get_test_mode()
model.summary()

opt = Adam()
model.compile(optimizer=opt, loss=losses.binary_crossentropy )

import numpy as np
X = np.random.randint(255, size=(100, 10, 10))
y = np.random.randint(2, size=(100))
 
    
model.fit(X, y, epochs=3, batch_size=10)
  

**Other info / logs**  
Epoch 1/3
100/100 [==============================] - 2s 21ms/step - loss: nan
Epoch 2/3
100/100 [==============================] - 0s 312us/step - loss: nan
Epoch 3/3
100/100 [==============================] - 0s 312us/step - loss: nan
